### ISSUE TITLE
🐛 BUG: `--experimental-local` broken on `pages dev`

### ISSUE THREAD

**Issue by WalshyDev** [cloudflare/workers-sdk#2637]
### Which Cloudflare product(s) does this pertain to?

Pages

### What version of `Wrangler` are you using?

2.8.1

### What operating system are you using?

Mac

### Describe the Bug

Running `--experimental-local` with `wrangler pages dev` will error out, it seems it can't handle our asset binding correctly right now

```
$ wrangler --version
 ⛅️ wrangler 2.8.1
-------------------

$ wrangler pages dev public --experimental-local
🚧 'wrangler pages <command>' is a beta command. Please report any issues to https://github.com/cloudflare/wrangler2/issues/new/choose
▲ [WARNING] No compatibility_date was specified. Using today's date: 2023-01-29.

  Pass it in your terminal:
  ` ``
  --compatibility-date=2023-01-29
  ` ``
  See https://developers.cloudflare.com/workers/platform/compatibility-dates/ for more information.


Compiling worker to "/var/folders/lc/vxf8prjn4kl4swcvhk9r8hrc0000gn/T/functionsWorker-0.6647145828727512.mjs"...
✨ Compiled Worker successfully
[NPXI] @miniflare/tre not available locally. Attempting to use npx to install temporarily.
[NPXI] Installing... (npx --prefer-offline -y -p @miniflare/tre@3.0.0-next.8)
[NPXI] Installed into /.npm/_npx/f763b2efd540e32a/node_modules.
[NPXI] To skip this step in future, run: npm install --save-dev @miniflare/tre@3.0.0-next.8
✘ [ERROR] local worker: TypeError: generateASSETSBinding2 is not a function

      at transformMf2OptionsToMf3Options
  (/node_modules/wrangler/wrangler-dist/cli.js:124653:21)
      at startLocalWorker
  (/node_modules/wrangler/wrangler-dist/cli.js:124221:34)
```

Somewhere between #2268 and now it appears to have broken.

**Comment by mrluc** [comment]
Same here:

    ...
    ... TypeError: generateASSETSBinding2 is not a function ...
    ...

**Comment by GregBrimble** [comment]
Should be fixed as of [wrangler@2.12.1](https://github.com/cloudflare/workers-sdk/releases/tag/wrangler%402.12.1)!
